### PR TITLE
Add TribeBALMinterCoordinator

### PR DESCRIPTION
**Pull request by TomAFrench**
A potential alternative design to #1249 for discussion.

### CODE REVIEW

**Review by nventuro — COMMENTED** [review]
I've reviwed the build info and checked that it matches, and deployed.

File: pkg/governance-scripts/contracts/BaseCoordinator.sol [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+        require(currentStage < getStagesLength(), "All stages completed");
+
+        _coordinatorStages[_currentStage]();
+
+        _advanceCurrentStage(currentStage);
+    }
+
+    function _getTimeSinceLastStageActivation() internal view returns (uint256) {
+        return block.timestamp - getStageActivationTime(_currentStage - 1);
+    }
```
[the remark this answers is not in the document]
Comment (TomAFrench) [reply pkg/governance-scripts/contracts/BaseCoordinator.sol]: Yeah, I wasn't too worried about this as this function is meaningless for the first stage. We'd either have to return `block.timestamp`, `0` or revert in a nicer way but either way if you're using it there then you're doing something wrong and if governance approves it then they're silly and deserve to vote on a fixed coordinator.

File: pkg/governance-scripts/contracts/BaseCoordinator.sol [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+interface ICurrentAuthorizer is IAuthorizer {
+    // solhint-disable-next-line func-name-mixedcase
+    function DEFAULT_ADMIN_ROLE() external view returns (bytes32);
+
+    function grantRole(bytes32 role, address account) external;
+
+    function revokeRole(bytes32 role, address account) external;
+
+    function renounceRole(bytes32 role, address account) external;
+}
```
[the remark this answers is not in the document]
Comment (TomAFrench) [reply pkg/governance-scripts/contracts/BaseCoordinator.sol:37]: Yes but I'm considering moving it out as it'll likely cause issues when it comes time to migrate the Authorizer.

File: pkg/governance-scripts/contracts/BaseCoordinator.sol (hunk 1 above)
[the remark this answers is not in the document]
Comment (nventuro) [reply pkg/governance-scripts/contracts/BaseCoordinator.sol]: Can we get a nicer revert? It'll make debugging easier :upside_down_face:

File: pkg/governance-scripts/contracts/20220606-tribe-bal-minter/README.md [hunk 3]
```diff
@@ -5,3 +5,7 @@ This contract mints the amount of BAL which would be claimable by the Tribe's [V
 ## Governance proposal
 
 <https://forum.balancer.fi/t/tribe-dao-unclaimable-bal-rewards/3196>
+
+## Deployment
+
+[Deployment task here](../../deployments/tasks/TribeBALMinterCoordinator)
```
Comment (nventuro) [inline pkg/governance-scripts/contracts/20220606-tribe-bal-minter/README.md:11]:
```suggestion
[Deployment task here](../../deployments/tasks/20220606-tribe-bal-minter-coordinator)
```

**Review by nventuro — APPROVED** [review]

**Pull request merged**